pcl_ros/CMakeLists.txt

```diff
@@ -1,245 +1,89 @@
-)
-target_link_libraries(pcl_ros_segmentation pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-add_dependencies(pcl_ros_segmentation ${PROJECT_NAME}_gencfg)
-class_loader_hide_library_symbols(pcl_ros_segmentation)
-
-## Declare the pcl_ros_surface library
-add_library (pcl_ros_surface
-  src/pcl_ros/surface/surface.cpp
-  # Compilation is much faster if we include all the following CPP files in surface.cpp
-  src/pcl_ros/surface/convex_hull.cpp
-  src/pcl_ros/surface/moving_least_squares.cpp
-)
-target_link_libraries(pcl_ros_surface ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-add_dependencies(pcl_ros_surface ${PROJECT_NAME}_gencfg)
-class_loader_hide_library_symbols(pcl_ros_surface)
-
-## Tools
-
-add_executable(pcd_to_pointcloud tools/pcd_to_pointcloud.cpp)
-target_link_libraries(pcd_to_pointcloud ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-add_executable(pointcloud_to_pcd tools/pointcloud_to_pcd.cpp)
-target_link_libraries(pointcloud_to_pcd ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-add_executable(bag_to_pcd tools/bag_to_pcd.cpp)
-target_link_libraries(bag_to_pcd pcl_ros_tf ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-add_executable(convert_pcd_to_image tools/convert_pcd_to_image.cpp)
-target_link_libraries(convert_pcd_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-add_executable(convert_pointcloud_to_image tools/convert_pointcloud_to_image.cpp)
-target_link_libraries(convert_pointcloud_to_image ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
-
-## Downloads
-
-catkin_download(table_scene_lms400.pcd http://download.ros.org/data/pcl/table_scene_lms400.pcd
-  DESTINATION ${CMAKE_CURRENT_SOURCE_DIR}/samples/data
-  MD5 546b5b4822fb1de21b0cf83d41ad6683
-)
-add_custom_target(download ALL DEPENDS download_extra_data)
+target_include_directories(pcl_ros_tf
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+    ${PCL_INCLUDE_DIRS})
+target_link_libraries(pcl_ros_tf ${PCL_LIBRARIES})
+ament_export_libraries(pcl_ros_tf)
 
 #############
 ## Testing ##
 #############
 
-if(CATKIN_ENABLE_TESTING)
-  find_package(rostest REQUIRED)
-  add_rostest_gtest(test_tf_message_filter_pcl tests/test_tf_message_filter_pcl.launch src/test/test_tf_message_filter_pcl.cpp)
-  target_link_libraries(test_tf_message_filter_pcl ${catkin_LIBRARIES} ${GTEST_LIBRARIES})
-  add_rostest(samples/pcl_ros/features/sample_normal_3d.launch ARGS gui:=false)
-  add_rostest(samples/pcl_ros/filters/sample_statistical_outlier_removal.launch ARGS gui:=false)
-  add_rostest(samples/pcl_ros/filters/sample_voxel_grid.launch ARGS gui:=false)
-  add_rostest(samples/pcl_ros/segmentation/sample_extract_clusters.launch ARGS gui:=false)
-  add_rostest(samples/pcl_ros/surface/sample_convex_hull.launch ARGS gui:=false)
-endif(CATKIN_ENABLE_TESTING)
+if(BUILD_TESTING)
+  find_package(ament_cmake_gtest REQUIRED)
+  # TODO
```

Comment:
Add the auto ament linters please

Reply:
I've added them, but is your intent for me to make the linter clean as well?

In general when doing a port like this I like to keep the changes as minimal as possible so we don't overload the ROS->ROS2 diffs with a whole bunch of formatting and whitespace changes. Perhaps making the linter clean is a final step in the port once everything else is working?